### DISCUSSION TITLE
Is it possible to watch $refs or DOM ?

### DISCUSSION THREAD

**Discussion by kohheepeace** [alpinejs/alpine#3620] [1. Help]
Hello, I am currently looking for a method to **detect DOM src changes** in a turbo-frame using turbo-frame with Alpinejs.

[Question]
- Is it possible to detect DOM attributes (in this case, src) in Alpinejs?
- Or, do I need to use [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) ?

This is the code which is not working as I expected now...
```erb
<div class="wp-block-latest-courses alignwide">
  <turbo-frame
    x-data="{
      page: new URLSearchParams(location.search).get('page') || 1
    }"
    :src="`/courses?page=${page}`"
    x-init="$watch('$el.src',(value) => {
      console.log(value); // 👈 here nothing happend...
      })"
  >
    Loading...
    <span x-text="page"></span>
  </turbo-frame>
</div>
```

The screen on which this code is executed looks like this.

Turbo automatically changes the src if you click a tag inside `turbo-frame`.
![2023-06-13 12 23 14](https://github.com/alpinejs/alpine/assets/29557494/e8bd341a-7244-485e-8714-b39f1a07ce61)

**Comment by kohheepeace** [comment]
Ref: https://github.com/alpinejs/alpine/discussions/2134#discussioncomment-1384380

**Comment by ekwoka** [comment]
Reactive effects require a Proxy object being used for interactions to pass through.

So native classes that have internal mutability or can be modified directly without going through a Proxy cannot be watched. So like File, Set, Map, Node, Element, etc.

You can use a mutation observer, or more likely turbo emits an event on the element you can listen for.

Depending on your needs, I actually have a plugin for Alpine that behaves similarly to this (I think, idk much about Turbo Frame). It's on npm as `@ekwoka/x-ajax` and on github at https://github.com/ekwoka/alpine-plugins

**Comment by kohheepeace** [comment]
Thanks for reply. 

I understand the syntax like this is not possible.
```html
<div
  x-init="$watch('$el.src',(value) => {
    console.log(value);
    })
>
...
</div>
```

**Comment by kohheepeace** [comment]
> You can use a mutation observer, or more likely turbo emits an event on the element you can listen for.

=> I finally use the code like below to listen the turbo event.

```html
<div class="wp-block-latest-courses alignwide">
  <turbo-frame
    x-data
    id="turbo-cof-latest-courses"
    src="/courses"
    @turbo:before-frame-render.window="
      window.scrollTo({ top: 0, behavior: 'smooth' });
    "
  >
    Loading...
  </turbo-frame>
</div>
```

I think this solution is not related Alpinejs directly. Sorry.
